datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -1239,15 +1239,10 @@ pub fn project(
         }
     }
     validate_unique_names("Projections", projected_expr.iter())?;
-    let input_schema = DFSchema::new_with_metadata(
-        exprlist_to_fields(&projected_expr, &plan)?,
-        plan.schema().metadata().clone(),
-    )?;
```

Comment:
it already was contained by `try_new_with_schema`, so we can use `try_new_with_schema` directly